tests/integration/src/xvm.rs

```diff
@@ -0,0 +1,334 @@
+
+/* Call EVM Payable:
+
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::env::{DefaultEnvironment, Environment};
+use ink::prelude::vec::Vec;
+
+#[ink::contract(env = CustomEnvironment)]
+mod call_xvm_payable {
+    use super::*;
+
+    #[ink(storage)]
+    pub struct CallXvmPayable {}
+
+    impl CallXvmPayable {
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self {}
+        }
+
+        #[ink(message, payable, selector = 42)]
+        pub fn call_xvm_payable(
+            &self,
+            target: Vec<u8>,
+            input: Vec<u8>,
+        ) -> CallResult {
+            let value = Self::env().transferred_value();
+            // Calling EVM
+            Self::env().extension().call(0x0F, target, input, value)
+        }
+    }
+}
+
+pub type CallResult = u32;
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The constant `CALL_WASM_PAYBLE` is defined as a string representation of WASM bytecode. It seems to be used for testing purposes. No issues found.



<!-- This is an auto-generated comment by OSS CodeRabbit -->